middleware/shell.go

```diff
@@ -0,0 +1,69 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os/exec"
+	"sync"
+	"time"
+)
+
+type Shell struct {
+	mu           sync.Mutex
+	Cmd          string
+	Args         []string
+	IntervalSecs int64
+	TimeoutSecs  int64
+}
+
+func NewShell(interval, timeout int64, cmd string, args ...string) (*Shell, error) {
+	if interval < 0 {
+		return &Shell{}, fmt.Errorf("shell needs to be set with a valid interval value")
+	}
+	if timeout < 0 {
+		return &Shell{}, fmt.Errorf("shell needs a valid timeout value")
+	}
+	return &Shell{
+		Cmd:          cmd,
+		Args:         args,
+		IntervalSecs: interval,
+		TimeoutSecs:  timeout,
+	}, nil
+}
+
+func (s *Shell) RunBg() *time.Ticker {
+	ticker := time.NewTicker(time.Duration(s.IntervalSecs) * time.Second)
+	go func() {
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				err := s.Run()
+				if err != nil {
+					log.Println("Failed to start cmd:", s.Cmd)
+					break
+				}
+			}
+		}
```

Comment:
Getting a `  Error: S1000: should use for range instead of for { select {} } (gosimple)` linter error which seems like a false positive for this use case here? Any suggestions @mjh1 @thomshutt?

Reply:
found this: https://github.com/dominikh/go-tools/issues/503
I think what it's really complaining about is the fact that's there's just a single case so I think it could just be replaced with my suggestion above

Reply:
good call - i think we can use the `range` style you suggested and get around this.